In [73]:
import pandas as pd

In [ ]:
# df1 = pd.read_csv('D:\\ML CODES\\IBM\\Test_Cleaned\\Apple_test_cleaned.csv', encoding='latin-1')

In [ ]:
# df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3725 entries, 0 to 3724
Data columns (total 3 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   sentiment             3725 non-null   int64  
 1   sentiment:confidence  3725 non-null   float64
 2   text                  3725 non-null   object 
dtypes: float64(1), int64(1), object(1)
memory usage: 87.4+ KB


In [76]:
# df1 = df1.drop(['_golden','_unit_state','_trusted_judgments','_last_judgment_at','date','id','query','sentiment_gold'], axis=1)

In [77]:
# df1 = df1.to_csv("Apple_test_cleaned.csv", index=False)

In [78]:
# df1.isnull().sum()

##### The 82 rows with not_relevant sentiment have been removed. The DataFrame now only contains the relevant sentiment values:



In [79]:
# df1 = df1[df1['sentiment'] != 'not_relevant']
# df1['sentiment'].value_counts()

##### 79 rows with fewer than 5 words in the text column were removed (3804 → 3725 rows remaining). 

In [80]:
# print("Before:", len(df1))
# df1 = df1[df1['text'].str.split().str.len() >= 5]
# print("After:", len(df1))

In [81]:
# df1.to_csv("Apple_test_cleaned2.csv", index=False)

In [ ]:
# df1 = pd.read_csv("D:\\ML CODES\\IBM\\Test_Cleaned\\hotel_test_cleaned.csv")

In [ ]:
# df2.head()

,review,sentiment
0,One of the other reviewers has mentioned that ...,positive
1,A wonderful little production. <br /><br />The...,positive
2,I thought this was a wonderful way to spend ti...,positive
3,Basically there's a family where a little boy ...,negative
4,"Petter Mattei's ""Love in the Time of Money"" is...",positive


In [ ]:
# df2 = df2[['reviews.text']]

In [ ]:
# df2.to_csv('hotel_test_cleaned.csv',index=False)

In [ ]:
# df3 = pd.read_csv("D:\\ML CODES\\IBM\\test\\socialmedia-test.csv",encoding='latin-1')

In [216]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4815 entries, 0 to 4814
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   textID            3534 non-null   object 
 1   text              3534 non-null   object 
 2   sentiment         3534 non-null   object 
 3   Time of Tweet     3534 non-null   object 
 4   Age of User       3534 non-null   object 
 5   Country           3534 non-null   object 
 6   Population -2020  3534 non-null   float64
 7   Land Area (Km²)   3534 non-null   float64
 8   Density (P/Km²)   3534 non-null   float64
dtypes: float64(3), object(6)
memory usage: 338.7+ KB


In [ ]:
# df3 = df3[['text']]

In [240]:
df1

,text,sentiment
0,Last session of the day,neutral
1,Shanghai is also really exciting precisely s...,positive
2,Recession hit Veronique Branquinho she has to ...,negative
3,happy bday,positive
4,I like it,positive
...,...,...
3529,its at 3 am im very tired but i cant sleep bu...,negative
3530,All alone in this old house again Thanks for ...,positive
3531,I know what you mean My little dog is sinking...,negative
3532,_sutra what is your next youtube video gonna b...,positive


In [238]:
print("df2 duplicates:", df1.duplicated().sum())


df2 duplicates: 0


In [237]:
df1.drop_duplicates(inplace=True)

In [ ]:
# df1 = df1[['text','sentiment']]

In [ ]:
# df1.isnull().sum()

text         0
sentiment    0
dtype: int64

In [ ]:
# df1 = df1.drop(['Id'],axis = 1)

##### Remove URLs

In [232]:
df1['text'] = df1['text'].str.replace(r"http\S+|www\S+", "", regex=True)

##### Remoev @ mentions


In [233]:
import re

df1["text"] = df1["text"].apply(lambda x: re.sub(r"@\w+", "", x))

##### Remove Punctuations

In [234]:
df1["text"] = df1["text"].apply(lambda x: re.sub(r"[^\w\s]", "", x))

#### handle elongated words

In [235]:


def remove_elongated_words(text):
    return re.sub(r'(.)\1{2,}', r'\1\1', text)

df1["text"] = df1["text"].apply(remove_elongated_words)

In [213]:
df1.to_csv("text_test_cleaned.csv", index=False)

In [ ]:
# df1 = pd.read_csv('D:\\ML CODES\\IBM\\test\\test.csv')

In [219]:
df1 = pd.read_csv('D:\\ML CODES\\IBM\\test\\test.tweet.csv',encoding='latin-1')